src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -362,36 +361,53 @@ else if (accidentallySerializable)
         // Define the generated class in this VM.
 
         final byte[] classBytes = cw.toByteArray();
-        // If requested, dump out to a file for debugging purposes
-        if (dumper != null) {
-            AccessController.doPrivileged(new PrivilegedAction<>() {
-                @Override
-                public Void run() {
-                    dumper.dumpClass(lambdaClassName, classBytes);
-                    return null;
-                }
-            }, null,
-            new FilePermission("<<ALL FILES>>", "read, write"),
-            // createDirectories may need it
-            new PropertyPermission("user.dir", "read"));
-        }
         try {
             // this class is linked at the indy callsite; so define a hidden nestmate
             Lookup lookup;
-            if (useImplMethodHandle) {
-                lookup = caller.defineHiddenClassWithClassData(classBytes, implementation, !disableEagerInitialization,
-                                                               NESTMATE, STRONG);
-            } else {
-                lookup = caller.defineHiddenClass(classBytes, !disableEagerInitialization, NESTMATE, STRONG);
+            try {
+                if (useImplMethodHandle) {
+                    lookup = caller.defineHiddenClassWithClassData(classBytes, implementation, !disableEagerInitialization,
+                        NESTMATE, STRONG);
+                } else {
+                    lookup = caller.defineHiddenClass(classBytes, !disableEagerInitialization, NESTMATE, STRONG);
+                }
+            } catch (Throwable t) {
```

Comment:
suggest to use try-finally in writing the file.

```
            Lookup lookup = null;
            try {
                if (useImplMethodHandle) {
                    lookup = caller.defineHiddenClassWithClassData(classBytes, implementation, !disableEagerInitialization,
                                                                   NESTMATE, STRONG);
                } else {
                    lookup = caller.defineHiddenClass(classBytes, !disableEagerInitialization, NESTMATE, STRONG);
                }
            } finally {
                // If requested, dump out to a file for debugging purposes
                if (dumper != null) {
                    final String name;
                    if (lookup != null) {
                        String definedName = lookup.lookupClass().getName();
                        int suffixIdx = definedName.lastIndexOf('/');
                        assert suffixIdx != -1;
                        name = lambdaClassName + '.' + definedName.substring(suffixIdx + 1);
                    } else {
                        name = lambdaClassName + ".failed-" + counter.incrementAndGet();
                    }
                    doDump(name, classBytes);
                }
            }
            return lookup.lookupClass();
```

Reply:
Done, with the small difference that I move the `return lookup.lookupClass()` into the `try` block because IntelliJ IDEA thought that `lookup` could be `null` at the point of the `return` statement; I figure, if IDEA can get that wrong then a casual reader might get it wrong too.